### PR TITLE
Introduce `:overrides` option

### PR DESCRIPTION
**Pull request by vemv**
## Brief

Implements https://github.com/nedap/formatting-stack/issues/134#issuecomment-590701727

Starting from this PR, `:overrides` becomes the primary API for customizing things: as a consumer, one should:

* Use one of: `git-status-formatter`, `branch-formatter`, `project-formatter`
* Pass :overrides to those if wanting to tweak something.

...and should not:

* Use `core/format!` directly
* Use the "factory" stuff introduced in this PR
 
## TODO 

* At this point, `git-status-formatter`, `branch-formatter`, `project-formatter` all deserve some test coverage.
* Explore how much can we DRY `git-status-formatter`, `branch-formatter`, `project-formatter`
  * A nice 'overrides' implementation would allow absolute DRY.
  * Might be less prioritary for now
* Create `impl` sub-nses for `git-status-formatter`, `branch-formatter`, `project-formatter` 
  * The intended API is `format-and-lint!` /  `lint!`
   * Things like `speced/defn default-formatters` are certainly not public
* Review doc (README)
* Review examples (test-resources)

## QA plan

* `git-status-formatter`, `branch-formatter`, `project-formatter` all keep working
* The examples work as advertised, customizing things as its code shows

## Author checklist

* [ ] I have QAed the functionality
* [ ] The PR has a reasonably reviewable size and a meaningful commit history
* [ ] I have run the [branch formatter](https://github.com/nedap/formatting-stack/blob/332a419034ab46fad526a5592f4257353bd695b6/src/formatting_stack/branch_formatter.clj) and observed no new/significative warnings
* [ ] The build passes
* [ ] I have self-reviewed the PR prior to assignment
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [ ] Correctness
  * [ ] Robustness (red paths, failure handling etc)
  * [ ] Modular design
  * [ ] Test coverage
  * [ ] Spec coverage
  * [ ] Documentation
  * [ ] Security
  * [ ] Performance
  * [ ] Breaking API changes
  * [ ] Cross-compatibility (Clojure/ClojureScript)

## Reviewer checklist

* [ ] I have checked out this branch and reviewed it locally, running it
* [ ] I have QAed the functionality
* [ ] I have reviewed the PR
* Additionally, I have code-reviewed iteratively the PR considering the following aspects in isolation:
  * [ ] Correctness
  * [ ] Robustness (red paths, failure handling etc)
  * [ ] Modular design
  * [ ] Test coverage
  * [ ] Spec coverage
  * [ ] Documentation
  * [ ] Security
  * [ ] Performance
  * [ ] Breaking API changes
  * [ ] Cross-compatibility (Clojure/ClojureScript)

### CODE REVIEW

File: src/formatting_stack/impl/overrides.clj [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+(defn removing-deep-merge
+  "A deep-merge in which `nil` values express that the given key-value entry should be removed."
+  ([a]
+   (if-not (map? a)
+     a
+     (-> (into {}
+               (filter (fn [[k v]]
+                         (some? v)))
+               a)
+         (with-meta (meta a)))))
```
Comment (thumbnail) [inline src/formatting_stack/impl/overrides.clj]:
```clj
     (->> (remove (comp nil? second) a)
          (into (with-meta {} (meta a))))
```

Comment (vemv) [reply src/formatting_stack/impl/overrides.clj]:
```clj
(-> (into {}
          (filter (comp val some?))
          a)
    (with-meta (meta a)))
```

* prefer `val` to `second`
* Correct `comp` order (it's the opposite in transducers)
* `remove nil?` reads like a double negative. So, `filter some? `
* Conceptually `into` happens first and `with-meta` second. First things should be read earlier (top-to-bottom)

**Comment by vemv** [comment]
@thumbnail : Progressed quite a bit. Although the branch isn't pretty, the extended specs, examples, protocols, tests etc should make it quite clear the overall intended API.

You might want to take over the branch, as I see it it's 90% there.

File: src/formatting_stack/core.clj [hunk 2]
```diff
@@ -47,30 +48,51 @@
...
 
-(defn format! [& {:keys [strategies
-                         third-party-indent-specs
-                         formatters
-                         linters
-                         processors
-                         reporter
-                         in-background?]}]
+(speced/defn format! [& {:keys                                             [^vector? strategies
+                                                                            ;; XXX rename `formatters` to `formatter-factories`
```
Comment (vemv) [inline src/formatting_stack/core.clj:52]: (Important to address this XXX)

**Comment by vemv** [comment]
Closing so that I'll push a rebased version.

**Pull request closed without merging**